### PR TITLE
Move failed attacks to core, update combos to break on failed attacks

### PR DESCRIPTION
**Pull request by AzariahAeroborn**

### CODE REVIEW

**Comment by AzariahAeroborn** [comment]
When I went back to double-check that updating combos fixed the "bloodspiller not generating blood" I had initially after refactoring the successfulHit logic, I observed that the signature of the ComboEvent interface had been changed by these changes, since it's extending off of aoedamage now instead of off cast.  Updated the inheritance accordingly.  

The only consumer of ComboEvent outside of the combos module is GNB's Ammo module, which is not effected by the additional properties in the updated signature.

File: src/parser/core/modules/Combos.tsx [hunk 1]
```diff
@@ -18,16 +18,30 @@ const GCD_TIMEOUT_MILLIS = 15000
...
 }
 
+interface Hit {
+	id: number
+	instance: number
+	times: number
+	amount: number
+}
+
+export interface AoeEvent extends AbilityEvent {
```
Comment (ackwell) [inline src/parser/core/modules/Combos.tsx]: be good to add a note here and in the aoe module that this should be over there

Comment (AzariahAeroborn) [reply src/parser/core/modules/Combos.tsx]: Updates made.

**Comment by AzariahAeroborn** [comment]
Not sure what order you're going to want to resolve these in (or if the changes are simple enough that auto-merge will be able to handle them), but this PR now will have merge conflicts with #486 (RotationTable) and #490 (SAM combos)

**Pull request merged**